### DISCUSSION TITLE
api_info and authenticated requests

### DISCUSSION THREAD

**Discussion by gdmachado** [pyload/pyload#4029] [Help] [answered]
Hello, I'm trying to write a Downloader for a specific sharing website I use, and I'm unsure what to do regarding `api_info()`. The website provides an API and an endpoint with some detailed info about a file, but you can only call this specific endpoint when authenticated.

The issue I'm having is, `api_info` is a classmethod, and I'm unable to access the account cookies for a specific account from within this method because of this. I'd like to fetch info from this endpoint when authenticated, and fallback to regex parsing the HTML otherwise. Any tips appreciated!

**Comment by GammaC0de** [comment]
@gdmachado Can you share your code / provide a test account?

I would like to try to solve this issue.

**Comment by GammaC0de** [comment, marked as answer]
Fixed in #4044
